DEVELOPERS_README.dox

```diff
@@ -0,0 +1,180 @@
+
+#include <votca/tools/vec.h>
+
+using namespace std;
+using namespace votca::tools;
+
+BOOST_AUTO_TEST_SUITE(vec_test)
+
+
+BOOST_AUTO_TEST_CASE(test1){
+  vecv;
+  BOOST_CHECK_EQUAL(...);
+  BOOST_CHECK_EQUAL(...);
+:
+}
+BOOST_AUTO_TEST_CASE(test2){
+  vecv;
+  BOOST_CHECK_EQUAL(...);
+  BOOST_CHECK_EQUAL(...);
+:
+}
+:
+BOOST_AUTO_TEST_SUITE_END()
+\endcode
+ *    Replace the '...' and ':' with the appropriate syntax. 
+ *    For more info on which boost test macros to use refer to the boost documentation
+ * 6. To compile and test the code create a folder tools/build and run the following
+ *    commands: 
+ *    \code{.unparsed}
+cmake -DENABLE_TESTING=ON ../
+make 
+make test
+\endcode
+ *    Ensure you have an up to date version of cmake or use cmake3 
+ *
+ * \section CPP Codeing Style Guide
+ *
+ * VOTCA now uses the clang formatter to automatically ensure consistent style.
+ * The style is essentially the same, as what is used by google, with the 
+ * exception that types and = are aligned to make the code more readable. 
+ *
+ * To run clang on a file say file.cc I need to first be sure that the 
+ * .clang-format file is placed in the current working directory. It should
+ * have the following contents:
+ * \code{.unparsed}
+---
+# BasedOnStyle:  Google
+AccessModifierOffset: -1
+ConstructorInitializerIndentWidth: 4
+AlignConsecutiveAssignments: true
+AlignConsecutiveDeclarations: true
+AlignEscapedNewlinesLeft: true
+AlignTrailingComments: true
+AllowAllParametersOfDeclarationOnNextLine: true
+AlwaysBreakTemplateDeclarations: true
+AlwaysBreakBeforeMultilineStrings: true
+AllowShortIfStatementsOnASingleLine: true
+AllowShortLoopsOnASingleLine: true
+BreakBeforeBinaryOperators: false
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializersBeforeComma: false
+BinPackParameters: true
+ColumnLimit:     80
+ConstructorInitializerAllOnOneLineOrOnePerLine: true
+DerivePointerBinding: true
+ExperimentalAutoDetectBinPacking: false
+IndentCaseLabels: true
+MaxEmptyLinesToKeep: 1
+NamespaceIndentation: None
+ObjCSpaceBeforeProtocolList: false
+PenaltyBreakBeforeFirstCallParameter: 1
+PenaltyBreakComment: 60
+PenaltyBreakString: 1000
+PenaltyBreakFirstLessLess: 120
+PenaltyExcessCharacter: 1000000
+PenaltyReturnTypeOnItsOwnLine: 200
+PointerBindsToType: true
+SpacesBeforeTrailingComments: 2
+Cpp11BracedListStyle: true
+Standard:        Auto
+IndentWidth:     2
+TabWidth:        8
+UseTab:          Never
+BreakBeforeBraces: Attach
+IndentFunctionDeclarationAfterType: true
+SpacesInParentheses: false
+SpacesInAngles:  false
+SpaceInEmptyParentheses: false
+SpacesInCStyleCastParentheses: false
+SpaceAfterControlStatementKeyword: true
+SpaceBeforeAssignmentOperators: true
+ContinuationIndentWidth: 4
+...
+\endcode
+ * 
+ * Now I am prepared to enforce the style guide by typing:
+ *
+ * \code{.unparsed}
+clang-fromat -i -style=file file.cc
+\endcode
+ *
+ * The -i ensures it will make change to file.cc, -style=file ensures the
```

Comment:
`-i` has the big downside that is automatically modified file, which isn't what most developers want.